src/Formik.tsx

```diff
@@ -448,13 +451,16 @@ export class Formik<Values = {}, ExtraProps = {}> extends React.Component<
         });
       }
 
-      this.setState(prevState => ({
-        touched: setIn(prevState.touched, field, true),
-      }));
-
-      if (this.props.validateOnBlur) {
-        this.runValidations(this.state.values);
-      }
+      this.setState(
+        prevState => ({
+          touched: setIn(prevState.touched, field, true),
+        }),
+        () => {
+          if (this.props.validateOnBlur) {
+            this.runValidations(this.state.values);
+          }
```

Comment:
It does not matter here, but I moved it for good measure. I can also undo this part.

Reply:
I would move this back and save a render.

Reply:
I force-pushed now to remove it.